master/internal/api_experiment.go

```diff
@@ -47,14 +52,22 @@ func (a *apiServer) GetExperiments(
 	return resp, a.paginate(&resp.Pagination, &resp.Experiments, req.Offset, req.Limit)
 }
 
-// For the 4 following endpoints: activate, pause, cancel, and kill if the experiment is not in
-// memory we treat it as if the experiment does not exist as these operations would fail anyway
-// on experiments on terminal states. This would affect idempotent property of kill and cancel
-// operations
-
 func (a *apiServer) ActivateExperiment(
 	ctx context.Context, req *apiv1.ActivateExperimentRequest) (
 	resp *apiv1.ActivateExperimentResponse, err error) {
+	var rawExp []byte
+	var exp model.Experiment
+	if rawExp, err = a.m.db.ExperimentRaw(int(req.Id)); err != nil {
+		return nil, status.Errorf(codes.NotFound, "%s; experiment %d not found.", err.Error(), req.Id)
+	}
+	if err = json.Unmarshal(rawExp, &exp); err != nil {
+		return nil, status.Error(codes.Internal, "failed to unmarshal experiment")
+	}
+	if model.TerminalStates[exp.State] {
```

Comment:
leave the check to actors. catch actor 404 and translate into incompatible state